hac-node/agent/service.go

```
@@ -0,0 +1,372 @@
+	if requestData.ProposalId != 0 {
+		proposalInfo, err := s.getProposalInfoById(requestData.ProposalId)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+		response.Proposals = append(response.Proposals, proposalInfo)
+		c.JSON(http.StatusOK, response)
+		return
+	}
+	proposalTotal := uint64(0)
+	proposals := make([]Proposal, 0)
```

golangci-lint (1.62.2)

ineffectual assignment to proposals

(ineffassign)